src/main/java/com/example/demo/reservationComponent/domain/entities/Customer.java

```diff
@@ -0,0 +1,37 @@
+package com.example.demo.reservationComponent.domain.entities;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+
+import java.util.UUID;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Builder
+@Table(
+        name = "customer",
+        indexes = {
+            @Index(name = "idx_customer_reservation_id", columnList = "reservation_id")
+        }
+)
+public class Customer {
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
+
+    @Column(name = "reservation_id", nullable = false)
+    private UUID reservationId;
+
+    @Column(nullable = false)
+    private String firstName;
+
+    @Column(nullable = false)
+    private String lastName;
+
+    @Column(nullable = false)
```

Comment:
maybe here should be used names for columns (for example first_name, last_name, phone_number), because right now almost everywhere used _ for separating words ,  do not use here camelCase.
Example 
`@Column(name= "last_name" , nullable = false)
    private String lastName;`